publisher.rb

```diff
@@ -1,33 +1,51 @@
 require 'aws-sdk-sqs'
 require_relative 'load_aws'
 require_relative 'aws_sqs_client'
+require_relative 'q_it_errors'
 
 class Publisher
   
   attr_reader :sqs, :sleep_period, :queue_name
 
   def initialize(options)
     @sqs = AwsSQSClient.new.client
-    @sleep_period = valid_sleep_period?(options[1]) ? options[1].to_i : 5
     @queue_name = options[0]
+    @sleep_period = options[1]
+  end
+
+  def self.validate(options)
+    raise QItArgumentError, "QItArgumentError :: Incorrect number of arguments. Expected 2 got #{options.count}" if options.count != 2
+    
+    valid_queue_name?(options[0]) && valid_sleep_period?(options[1])
   end
 
   # For Standard Queue
   def create_std_q
     sqs.create_queue(queue_name: queue_name)
   end
 
-  def valid_sleep_period?(sleep_period)
+  def self.valid_sleep_period?(sleep_period)
     max_sleep_period = /\A([5-9]|1\d{1}|20)\z/
-    !sleep_period.nil? && sleep_period.match(max_sleep_period)
+    
+    raise QItInvalidArgumentError, "QItInvalidArgumentError :: #{self} Invalid sleep period #{sleep_period}. Sleep period range is between 5-20 seconds." unless sleep_period.match(max_sleep_period)
```

Comment:
Mentioned it earlier - adding in case it is missed later. Use `match?` and not `match`.